src/sentry/integrations/slack/tasks.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import absolute_import
+
+import six
+
+from django.conf import settings
+
+from sentry import http
+from sentry.utils import json
+from sentry.tasks.base import instrumented_task
+from sentry.models import Integration, Project, Rule
+
+from sentry.utils.redis import redis_clusters
+from sentry.mediators import project_rules
+
+MEMBER_PREFIX = "@"
+CHANNEL_PREFIX = "#"
+
+LIST_TYPES = [
+    ("channels", "channels", CHANNEL_PREFIX),
+    ("groups", "groups", CHANNEL_PREFIX),
+    ("users", "members", MEMBER_PREFIX),
+]
+
+
+class RedisRuleStatus(object):
+    def __init__(self, uuid):
+        self.uuid = uuid
+
+        cluster_id = getattr(settings, "SENTRY_RULE_TASK_REDIS_CLUSTER", "default")
```

Comment:
This logic seems repeated so consider putting this into a helper in a follow up patch?